frontend/src/lib/components/LemonModal/LemonModal.tsx

```diff
@@ -1,21 +1,99 @@
 import React from 'react'
-import { Modal, ModalProps } from 'antd'
 import { IconClose } from 'lib/components/icons'
 import { LemonButton } from 'lib/components/LemonButton'
-import clsx from 'clsx'
+import Modal from 'react-modal'
 import './LemonModal.scss'
 
-export type LemonModalProps = React.PropsWithChildren<Omit<ModalProps, 'closeIcon'>>
+export type LemonModalHeaderProps = {
+    children?: React.ReactNode
+}
+
+export type LemonModalFooterProps = {
+    children?: React.ReactNode
+}
+
+export interface LemonModalProps {
+    children?: React.ReactNode
+    isOpen: boolean
+    onClose: () => void
+    width?: number
+    inline?: boolean
+    title: string | JSX.Element
+    description?: string | JSX.Element
+    footer?: React.ReactNode
+    /** When enabled, the modal content will only include children allowing greater customisation */
+    simple?: boolean
+}
+
+export const LemonModalHeader = ({ children }: LemonModalHeaderProps): JSX.Element => {
+    return <header className="LemonModal__header">{children}</header>
+}
+
+export const LemonModalFooter = ({ children }: LemonModalFooterProps): JSX.Element => {
+    return <footer className="LemonModal__footer">{children}</footer>
+}
+
+export const LemonModalContent = ({ children }: LemonModalFooterProps): JSX.Element => {
+    return <section className="LemonModal__content">{children}</section>
+}
 
 /** A lightweight wrapper over Ant's Modal for matching Lemon style. */
-export function LemonModal({ className, footer = null, width = 480, ...modalProps }: LemonModalProps): JSX.Element {
-    return (
+export function LemonModal({
+    width,
+    children,
+    isOpen,
+    onClose,
+    title,
+    description,
+    footer,
+    inline,
+    simple,
+}: LemonModalProps): JSX.Element {
+    const modalContent = (
+        <>
+            <div className="LemonModal__closebutton">
+                <LemonButton icon={<IconClose />} size="small" status="stealth" onClick={onClose} />
```

Comment:
These workarounds with wrapping `div`s seem kind of pointless to me, this would be more straightforward if `LemonButton` had this as an extra `className` here.

Reply:
Mentioned this in the other PR. I feel split on this. On one side I want to push for as little deviation from the design as possible hence limiting classnames, on the other hand it makes these useful helpers like layout harder...